hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/ReadableKVStateBase.java

```diff
@@ -131,6 +137,7 @@ protected final void markRead(@NonNull K key, @Nullable V value) {
      * @return Whether it has been read
      */
     protected final boolean hasBeenRead(@NonNull K key) {
-        return readCache.containsKey(key);
+        var keyExistence = readCache.get(key);
+        return keyExistence != null && keyExistence != marker;
```

Comment:
Why do we need to check `keyExistence != marker;` ?

Reply:
Because we put a marker into the reacCache and we would like to return true just if it is not marker and keyExistance is not null.

Reply:
There is an interaction with line 72, so if this is changed, then line 72 cannot call this method and must implement the marker check.

Reply:
Agreed with @Neeharika-Sompalli , the implementation here needs to remain `readCache.containsKey(key)`.

Reply:
You are right. Sorry, did not think this through.

Reply:
fixed it